README.md

```diff
@@ -229,11 +231,17 @@ Example:
     AWS_COMMON_API
  void aws_module_destroy(aws_module_t *module);
 
-* Avoid c-strings, and don't write code that depends on `NULL` terminators. Expose `struct aws_byte_buf` APIs
-and let the user figure it out.
+* Avoid c-strings, and don't write code that depends on `NULL` terminators.
+    * Pass strings via `struct aws_byte_cursor`. This is a non-owning view type. Pass it by value.
+    * Only pass `const char *` when thinly wrapping an OS function that *requires* a `NULL` terminator.
```

Comment:
should this call out that cursors dont have null terminator after then and you need to call a function to get a char * with one